ChallengeTrackerAppAngular/src/app/userManagement/register-user/register-user.component.ts

```diff
@@ -0,0 +1,53 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { NgForm } from '@angular/forms';
+
+import { UserService } from "../../services/user.service";
+import { User } from "../../models/user";
+
+@Component({
+  selector: 'app-register-user',
+  templateUrl: './register-user.component.html',
+  styleUrls: ['./register-user.component.css']
+})
+export class RegisterUserComponent implements OnInit {
+  submitted: boolean;
+	newUser: User;
+  resultSuccess: boolean;
+	resultError: boolean;
+	message: string | undefined;
+
+  constructor(private router: Router,
+    private activatedRoute: ActivatedRoute,
+    private userService: UserService) 
+  { 
+    this.submitted = false;
+    this.newUser = new User();
+    this.resultSuccess = false;
+    this.resultError = false;
+  }
+
+  ngOnInit(): void {
+  }
+
+  create(createUserForm: NgForm){
+    this.submitted = true;
+    if(createUserForm.valid){
```

Comment:
for readability, take out the code inside the if block and return early if createUserForm is not valid

Reply:
done